### PR TITLE
US118543 [Engagement] Loading states

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US118543](https://rally1.rallydev.com/#/detail/userstory/404486515748?fdp=true): [Engagement] Loading states

AC:

Add the spinners/grey cards/table into the UI for when the dashboard is loading as per the design

- [x] Overlay with spinner
- [x] Keyboard navigation while loading
- [x] Screen reader announcement
  - [x] Get proper langterms from Kevin
- [x] Unit tests
- [x] Grey card/table - separate PR

Out of scope of the story:
* Tab-navigation over dashboard's cards is a part of US120050 story. I believe that tab-navigation when loading is in-progress should be part of that story as well.
* Grey cards/table - skeleton components are [in active development](https://d2l.slack.com/archives/C0PHG3QB0/p1600085625042300). Should be implemented in a separate PR or story. 

### Functional Testing
* Unit tests
* Manual tests  
  * Keyboard navigation
  * Announcement when loading is started/finished. 
    * Alerts are not accessible for screen reader via tab and arrow navigation.
    * Alerts are not visible on the screen
    * FF: 1) alert Loading is in progress; 2) Loading is finished alert. NVDA repeats Loading is finished
    * Chrome: 1) alert Loading is in progress; 2) Loading is finished alert. NVDA repeats Loading is finished
    * Edge: 1) alert Loading is in progress; 2) Loading is finished alert .
  * Screen reader reads only titles of summary cards if tabindex is set
  * Charts are not available for tab-navigation when data is loading
  * Screen reader

### UX and docs 
- [x] Demo to Kevin
- [x] NVDA
- [x] JAWS
- [x] FF 
- [x] Chrome
- [x] Edge
- [x] ~Legacy edge~ - N/A

loading

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
FYI @anhill-D2L @Vitalii-Misechko @rohitvinnakota @johngwilkinson

**Comment by bemailloux** [comment]
This looks really cool. It almost doesn't need the grey cards!

File: components/aria-loading-progress.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-insights-aria-loading-progress {
+				left: -100vw;
```
Comment (bemailloux) [inline components/aria-loading-progress.js:22]: Are you doing this to "hide" the element by displaying it outside of the screen? Is it possible to use `display: none` instead?

File: components/overlay.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			:host {
+				display: block;
+				left: 0;
+				position: absolute;
+				top: 0;
+			}
+			:host([loading]) {
+				height: 100%;
+				width: 100%;
+				z-index: 1;
```
Comment (bemailloux) [inline components/overlay.js:30]: Thanks for not using `z-index: 9999` :)

File: test/components/current-final-grade-card.test.js [hunk 3]
```diff
@@ -31,6 +31,20 @@ describe('d2l-insights-current-final-grade-card', () => {
 			expect(title[0].innerText).to.equal('Current Grade');
 			expect(el._preparedHistogramData.toString()).to.equal(data.currentFinalGrades.toString());
 		});
+
+		it('should exclude chart form tabindex when data is loading', async() => {
```
Comment (bemailloux) [inline test/components/current-final-grade-card.test.js:35]: Nice!

**Review by bemailloux — APPROVED** [review]

File: components/aria-loading-progress.js (hunk 1 above)
Comment (Vitalii-Misechko) [reply components/aria-loading-progress.js:22]: Yes, I'm.  `display: none` hides text from screen readers as well. we don't want' that

File: components/overlay.js (hunk 2 above)
Comment (Vitalii-Misechko) [reply components/overlay.js:30]: :D :D :D

**Review by MykolaGalian — APPROVED** [review]

**Pull request merged**